File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        // By Moores voting algorithms it cancel each other
        int el;
        int count=0;
        for(int i=0;i<nums.size();i++){
            if(count==0){
                el=nums[i];
                count++;
            }
            else if(nums[i]==el){
                count++;
            }
            else{
                count--;
            }
        }
        int count1=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==el){
                count1++;
            }
            if(count1>(nums.size()/2)){
                return el;
            }
        }
        return -1;
        
    }
};

File: Linked_List/169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {

        int n = nums.length;

        int curr=nums[0];
        int vote=1;

        for(int i=1;i<n;i++){


            if(nums[i]==curr){
                vote++;
            }
            else if(vote==0){
                curr=nums[i];
                vote=1;
            }
            else{
                vote--;
            }
        }
        return curr;
    }
}